### DISCUSSION TITLE
Resolving Missing pandoc Symlink in Quarto Installed via Pixi Environment

### DISCUSSION THREAD

**Discussion by laguill** [quarto-dev/quarto-cli#11593] [Q&A] [answered]
Hello,
I wanted to share an issue I faced while using Quarto in a Pixi environment on Ubuntu.

### The Issue
When Quarto is installed in a Pixi environment, the required pandoc symlink is missing. This causes issues with the Quarto VS Code extension, as the visual editing mode relies on pandoc being available at a specific path (/opt/quarto/bin/tools/pandoc).

### Solution Overview
To resolve this, you need to create a manual symlink for pandoc in the appropriate directory. Below is a step-by-step guide.

### Step-by-Step Solution
  1. **Determine System Architecture**
      Use the `uname -m` command to identify your system's architecture. This is necessary because the location for the `pandoc` binary depends on whether you're using `x86_64` (64-bit Intel/AMD) or `aarch64` (ARM).x86_64 architecture or aarch64 you have to add symlink manually
  2. **Use the**  **​`$CONDA_PREFIX`​** **Environment Variable**
      The `$CONDA_PREFIX` variable points to the root directory of your active Conda environment. We'll use it to construct the paths dynamically, ensuring compatibility across environments.
  3. **Create the Required Path**
      Before creating the symlink, ensure the target directory exists. Create the `tools/<architecture>` directory if it doesn't already exist:

      ```bash
      mkdir -p $CONDA_PREFIX/bin/tools/$(uname -m)/
      ```

      The `-p` flag ensures that intermediate directories are created if needed.
  4. **Create the Symlink**
      Link the `pandoc` binary from `$CONDA_PREFIX/bin` to the architecture-specific path under `tools`:

      ```bash
      ln -s $CONDA_PREFIX/bin/pandoc $CONDA_PREFIX/bin/tools/$(uname -m)/pandoc
      ```

      This ensures that tools looking for `pandoc` in the architecture-specific directory can find it.
  5. Commands summary

      ```bash
      # Determine system architecture
      uname -m

      # Create path to pandoc symlink before adding it
      mkdir -p $CONDA_PREFIX/bin/tools/$(uname -m)/

      # Create the symlink
      ln -s $CONDA_PREFIX/bin/pandoc $CONDA_PREFIX/bin/tools/$(uname -m)/pandoc
      ```

**Comment by mcanouil** [comment]
Could you elaborate on how you install Quarto exactly?

And are you reporting an issue or actually reporting a way to solve one for an unofficial install of Quarto? GitHub Issues are for bug reports, whilst GitHub Discussions are preferred for discussions. (Don't open a new one, we can convert/transfer)

Note that the official packaged version of Quarto are listed on GitHub release: https://github.com/quarto-dev/quarto-cli/releases/tag/v1.6.39

Conda recipe for instance is not (really) maintain by Quarto team.

For instance for the current stable release, the official and supported installers are:

- quarto-1.6.39-win.msi
- quarto-1.6.39-win.zip
- quarto-1.6.39-macos.tar.gz
- quarto-1.6.39-macos.pkg
- quarto-1.6.39-linux-rhel7-amd64.tar.gz
- quarto-1.6.39-linux-amd64.tar.gz
- quarto-1.6.39-linux-arm64.tar.gz
- quarto-1.6.39-linux-amd64.deb
- quarto-1.6.39-linux-arm64.deb
- quarto-1.6.39.tar.gz

**Comment by laguill** [comment]
Sorry
I installed quarto from the conda forge source using `pixi add quarto`

**Comment by laguill** [comment]
I'm new to Python environments and have been trying to figure out the best way to include Quarto in my project's dependencies. Quarto is essential for my workflow, as I use it for generating fmy reports.

Is there a reason why I shouldn't include Quarto as part of my environment dependencies (e.g., in the `requirements.txt` or `environment.yml`? 
Would doing so create potential issues with dependency management or workflow in Python projects? 
I want to ensure my setup is clean and reproducible. Thanks!

**Reply by mcanouil** [reply]
I personally don’t like or use Conda and prefer Virtual Env which in my case revealed to be quite reliable contrary to Conda.

have you consider trying the pip method to install Quarto CLI?

```sh
pip install quarto-cli
```

it might work better than Conda.

**Comment by mcanouil** [comment]
Let's move this to a discussion.

**Comment by cderv** [comment]
I did not know about pixi: https://pixi.sh/latest/

The tool is building on Conda ecosystem: https://pixi.sh/latest/vision/#conda

So I believe it will install the quarto bundle from conda-forge: https://anaconda.org/conda-forge/quarto

This means the symlink problem needs to be resolved at Conda forge packaging: https://github.com/conda-forge/quarto-feedstock

I am surprised this has not been reported yet there. To be sure it does not come from pixi itself, just using conda should be used. 

Anyhow, can you open an issue in https://github.com/conda-forge/quarto-feedstock as this is the installer you are using ? Thanks ! 

> I'm new to Python environments and have been trying to figure out the best way to include Quarto in my project's dependencies. Quarto is essential for my workflow, as I use it for generating fmy reports.

To add on that, 

* You can install quarto on your system using one of the installer. And then easily update. That is the easiest way to get started with Quarto.
* If you want to pin a version for some reason in a project, you can use virtual environment with Python, or conda and install it using the right tool. This is considered advanced usage. (Some of new tooling with python like **uv** are proven to be fast an reliable https://docs.astral.sh/uv/

Hope it helps

**Reply by laguill** [reply, marked as answer]
- For now as my study require to work with conda package I will stay with that.
Also Pixi is integrating uv see [uv in pixi](https://prefix.dev/blog/uv_in_pixi)

 - I installed quarto binary finally to reduce possible other issues
 - Issue is created [quarto-feedstock/issues/67](https://github.com/conda-forge/quarto-feedstock/issues/67)
